pkg/performanceprofile/controller/performanceprofile/hypershift/components/handler.go

```diff
@@ -0,0 +1,314 @@
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components/machineconfig"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components/manifestset"
+	profileutil "github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components/profile"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/hypershift"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/resources"
+)
+
+const (
+	hypershiftPerformanceProfileNameLabel = "hypershift.openshift.io/performanceProfileName"
+	hypershiftNodePoolLabel               = "hypershift.openshift.io/nodePool"
+	tunedConfigMapLabel                   = "hypershift.openshift.io/tuned-config"
+	tunedConfigMapConfigKey               = "tuning"
+	mcoConfigMapConfigKey                 = "config"
+	ntoGeneratedMachineConfigLabel        = "hypershift.openshift.io/nto-generated-machine-config"
+)
+
+var _ components.Handler = &handler{}
+
+type handler struct {
+	controlPlaneClient client.Client
+	dataPlaneClient    client.Client
+	scheme             *runtime.Scheme
+}
+
+func NewHandler(controlPlaneClient, dataPlaneClient client.Client, scheme *runtime.Scheme) components.Handler {
+	return &handler{
+		controlPlaneClient: controlPlaneClient,
+		dataPlaneClient:    dataPlaneClient,
+		scheme:             scheme,
+	}
+}
+
+func (h *handler) Delete(ctx context.Context, profileName string) error {
+	// ConfigMap is marked for deletion and waiting for finalizers to be empty
+	// so better to clean up and delete the objects.
+
+	// delete RtClass, as right now all the other components created from this PP
+	// are embedded into ConfigMaps, which has an OwnerReference with the PP configmap
+	// and will be deleted by k8s machinery triggering the deletion procedure of the embedded elements.
+	rtName := components.GetComponentName(profileName, components.ComponentNamePrefix)
+	rtClass, err := resources.GetRuntimeClass(ctx, h.dataPlaneClient, rtName)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			// rtClass not found so nothing to delete, so delete process has finished
+			return nil
+		}
+		return fmt.Errorf("unable to read RuntimeClass %q, error: %w. Unable to finalize deletion procedure properly", rtName, err)
+	}
+
+	err = h.dataPlaneClient.Delete(ctx, rtClass)
+	return err
+}
+
+func (h *handler) Exists(ctx context.Context, profileName string) bool {
+	operatorNamespace := config.OperatorNamespace()
+	tunedName := components.GetComponentName(profileName, components.ProfileNamePerformance)
+	if _, err := resources.GetTuned(ctx, h.controlPlaneClient, tunedName, operatorNamespace); !k8serrors.IsNotFound(err) {
+		klog.Infof("Tuned %q is still exists under the namespace %q", tunedName, operatorNamespace)
```

Comment:
nit: "under the namespace" reads clunky, I believe better english is "in the namespace"